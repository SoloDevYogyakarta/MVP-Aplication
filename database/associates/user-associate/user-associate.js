"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.userHistoryInclude = exports.userInclude = exports.userAttribute = exports.userAssociate = void 0;
const history_entity_1 = require("../../../database/entities/authenticate/history-entity/history-entity");
const user_entity_1 = require("../../../database/entities/authenticate/user-entity/user-entity");
const order_associate_1 = require("../order-associate/order-associate");
const userAttribute = {
    include: [
        ['createdAt', 'created_at'],
        ['updatedAt', 'updated_at'],
    ],
    exclude: ['password', 'createdAt', 'updatedAt'],
};
exports.userAttribute = userAttribute;
const userInclude = [
    {
        model: history_entity_1.userHistoryEntity,
        attributes: {
            include: [['updatedAt', 'updated_at']],
            exclude: [],
        },
        as: 'activities',
    },
];
exports.userInclude = userInclude;
const userHistoryInclude = [
    {
        model: order_associate_1.orderAssociate,
        attributes: order_associate_1.orderAttribute,
        include: order_associate_1.orderInclude,
        as: 'order',
    },
];
exports.userHistoryInclude = userHistoryInclude;
user_entity_1.userEntity.hasMany(history_entity_1.userHistoryEntity, {
    sourceKey: 'id',
    foreignKey: { name: 'user_id', allowNull: true },
    as: 'activities',
});
user_entity_1.userEntity.hasMany(order_associate_1.orderAssociate, {
    sourceKey: 'id',
    foreignKey: { name: 'user_id', allowNull: true },
    as: 'order',
});
const userAssociate = user_entity_1.userEntity;
exports.userAssociate = userAssociate;
//# sourceMappingURL=user-associate.js.map