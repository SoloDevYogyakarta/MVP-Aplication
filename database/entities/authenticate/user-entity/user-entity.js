"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.userEntity = exports.UserEntity = void 0;
const sequelize_1 = require("sequelize");
const base_entity_1 = require("../../base-entity/base-entity");
const entity_1 = require("../../entity");
class UserEntity extends base_entity_1.BaseEntity {
}
exports.UserEntity = UserEntity;
exports.userEntity = entity_1.sequelize.define('USER', {
    plat_number: sequelize_1.DataTypes.STRING,
    phone_number: sequelize_1.DataTypes.STRING,
    name: sequelize_1.DataTypes.STRING,
    motor: sequelize_1.DataTypes.STRING,
    year_production: sequelize_1.DataTypes.INTEGER,
    address: sequelize_1.DataTypes.STRING,
    role: sequelize_1.DataTypes.STRING,
    password: sequelize_1.DataTypes.STRING,
}, {
    tableName: 'USER',
    schema: 'AUTHENTICATE',
});
//# sourceMappingURL=user-entity.js.map