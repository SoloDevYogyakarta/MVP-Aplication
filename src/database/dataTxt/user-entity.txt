{"id":35,"public_id":"F_STaX3k0xnN4rHA_hzx5","username":"updated-Ernestine39","email":"Alison_Blanda@hotmail.com","phone_number":null,"password":"$2b$10$LcKUdoFuBbJKdKnfin1L9.02XRry1L5.Jr4v/Fqq7qv8HNwXhrCHq","file_id":"YcJo1tS7b22wKYOZlyt3U","createdAt":"2023-10-22T07:49:37.093Z","updatedAt":"2023-10-22T07:49:37.168Z"}