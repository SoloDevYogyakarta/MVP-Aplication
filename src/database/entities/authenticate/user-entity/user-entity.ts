import { nanoid } from 'nanoid';
import { DataTypes, Model, ModelAttributes, Optional } from 'sequelize';
import { SuperTest } from 'supertest';
import { BaseEntity } from '../../base-entity/base-entity';
import { sequelize } from '../../entity';

export class UserEntity extends BaseEntity {
  plat_number: string;
  phone_number: string;
  name: string;
  motor: string;
  year_production: number;
  address: string;
  role: string;
  password: string;
}

export type UserCreationAttribute = Optional<UserEntity, 'id'>;
export interface UserInstance
  extends Model<UserCreationAttribute, UserEntity>,
    UserEntity {}
export const userEntity = sequelize.define<UserInstance>(
  'USER',
  {
    plat_number: DataTypes.STRING,
    phone_number: DataTypes.STRING,
    name: DataTypes.STRING,
    motor: DataTypes.STRING,
    year_production: DataTypes.INTEGER,
    address: DataTypes.STRING,
    role: DataTypes.STRING,
    password: DataTypes.STRING,
  } as ModelAttributes<UserInstance>,
  {
    tableName: 'USER',
    schema: 'AUTHENTICATE',
  },
);
