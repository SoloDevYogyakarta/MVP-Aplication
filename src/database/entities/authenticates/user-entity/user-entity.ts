import { DataTypes, Model, ModelAttributes, Optional } from 'sequelize';
import { nanoid } from 'nanoid';
import bcrypt from 'bcrypt';
import { sequelize } from '../../entity';
import { BaseEntity } from '../../base-entity/base-entity';

export class UserEntity extends BaseEntity {
  username: string;
  email: string;
  phone_number: number;
  password: string;
  role: string;
  file_id: string;
}

export type UserCreationAttribute = Optional<UserEntity, 'id'>;
export interface UserInstance
  extends Model<UserCreationAttribute, UserEntity>,
    UserEntity {}
export const userEntity = sequelize.define<UserInstance>(
  'USER',
  {
    public_id: DataTypes.STRING,
    username: DataTypes.STRING,
    email: DataTypes.STRING,
    phone_number: DataTypes.INTEGER,
    role: DataTypes.STRING,
    password: DataTypes.STRING,
    file_id: DataTypes.STRING,
  } as ModelAttributes<UserInstance>,
  {
    tableName: 'USER',
    schema: 'AUTHENTICATE',
    hooks: {
      beforeCreate(attributes, options) {
        const instance = attributes;
        instance.public_id = nanoid();
        instance.password = bcrypt.hashSync(instance.password, 10);
      },
    },
  },
);
