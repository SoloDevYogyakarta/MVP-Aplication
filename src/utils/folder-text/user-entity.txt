{"id":1057,"plat_number":"B 6929 SHA","phone_number":"540-311-3034","name":"Yvette Tremblay","motor":"Sleek Soft Mouse","address":"39389 Franey Courts","year_production":2861,"role":"admin","password":"$2b$10$Q3pPGa3yL/6IFbR/nUsj8uq2hFGtAbDTUkMqX5A.FyonjCXibU58K","updatedAt":"2023-11-06T15:51:08.666Z","createdAt":"2023-11-06T15:51:08.666Z"}