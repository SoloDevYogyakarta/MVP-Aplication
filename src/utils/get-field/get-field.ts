import { GetFieldType } from 'src/types/get-field.type';
import { readpath } from '../system/system';

export function getField(filename?: GetFieldType) {
  try {
    if (filename === 'fake') {
      return readpath(`../../database/dataTxt/${filename}.txt`);
    }
    return JSON.parse(readpath(`../../database/dataTxt/${filename}.txt`));
  } catch (err) {
    throw new Error(err);
  }
}
