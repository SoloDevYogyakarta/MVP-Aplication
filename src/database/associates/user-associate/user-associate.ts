import { FindAttributeOptions, Includeable } from 'sequelize';
import { userEntity } from '../../../database/entities/authenticates/user-entity/user-entity';
import { fileUserAssociate } from '../file-associate/file-associate';

const userAttribute: FindAttributeOptions = {
  include: [],
  exclude: [],
};

const userInclude: Includeable[] = [
  {
    model: fileUserAssociate,
    as: 'file',
  },
];

userEntity.belongsTo(fileUserAssociate, {
  targetKey: 'public_id',
  foreignKey: {
    name: 'file_id',
  },
  as: 'file',
});

const userAssociate = userEntity;
export { userAttribute, userInclude, userAssociate };
