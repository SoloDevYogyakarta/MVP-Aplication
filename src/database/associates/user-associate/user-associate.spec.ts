import { ModelCtor } from 'sequelize';
import { UserInstance } from '../../../database/entities/authenticates/user-entity/user-entity';
import { getField } from '../../../utils/get-field/get-field';
import { userAssociate } from './user-associate';

describe('userAssociate', () => {
  let entity: ModelCtor<UserInstance>;

  beforeEach(() => {
    entity = userAssociate;
  });

  it('should to be defined', () => expect(entity).toBeDefined());

  it('render correctly', () => expect(entity).toMatchSnapshot());

  try {
    it('findOne with relationship', async () => {
      const { public_id } = getField('user-http-entity');
      const findOne = await entity.findOne({ where: { public_id } });
      expect(findOne.public_id).toEqual(public_id);
    });

    it('findAll with relationship', async () => {
      const findAll = await entity.findAll();
      expect(findAll.length).not.toEqual(0);
    });
  } catch (err) {
    // empty
  }
});
