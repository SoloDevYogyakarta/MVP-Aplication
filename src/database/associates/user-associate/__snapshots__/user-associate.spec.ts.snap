// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`userAssociate render correctly 1`] = `[Function]`;
